packages/rest/src/rest.ts

```diff
@@ -172,6 +304,31 @@ export interface ResourceOperation {
 
 const resourceOperationsKey = createStateSymbol("resourceOperations");
 
+function resourceRouteProducer(
+  program: Program,
+  operation: Operation,
+  parentSegments: string[],
+  overloadBase: HttpOperation | undefined,
+  options: RouteOptions
+): DiagnosticResult<RouteProducerResult> {
+  // NOTE: The purpose of this producer is to pass along the behavior of the
+  // DefaultRouteProducer while setting the appropriate HTTP verb based on any
+  // resource operation decorators that have been applied.  This behavior will
+  // be overridden by the `autoRouteProducer` if `autoRoute` is also applied to
+  // the same operation.
+
+  // Set the OperationVerbSelector to pick verbs based on resource operation type
+  const paramOptions = {
+    ...(options?.paramOptions ?? {}),
+    verbSelector: getResourceOperationHttpVerb,
+  };
+
+  return DefaultRouteProducer(program, operation, parentSegments, overloadBase, {
```

Comment:
Disregard earlier question about exporting this. I see the utility now. 

This is a nice pattern. :)

Reply:
Thanks!  It took a few iterations, but I think it gives us the flexibility we need now